### PR TITLE
Add categories to site

### LINKED ISSUE #162: Create infrastructure to sort posts by type
Now that we've updated the number of types there are for posts (#133) I think it'd be nice if we used those types to sort the posts. It was discussed a little bit [here](https://github.com/tholman/inspiring-online/issues/133#issuecomment-192291643) but this can be a place to discuss it more in detail.

### PR DESCRIPTION
**Pull request by TJScalzo**
**Reminder:** This is not completely done. I still need to add the image for the introductory post. 
### I would love some feedback.
(Fixes #162)

### CODE REVIEW

**Comment by Haroenv** [comment]
Kinda hard to preview, because it's relying on *`/`-root-type links* and thus your fork doesn't load the css. I do like the idea and it looks pretty well worked out.

**Comment by TJScalzo** [comment]
@Haroenv Should I change the way the links work? What would I change them to?

**Comment by Haroenv** [comment]
Well it works well for the production because then `inspiring.online` is the root, making all links relative would be annoying

**Comment by TJScalzo** [comment]
I'm not quite sure I understand what you've said @Haroenv. I made the links so that they weren't hardcoded to link to `inspiring.online/...` I made them so that they would also work with `localhost:4000/...` Is that sort of what you're talking about?

**Comment by Haroenv** [comment]
Yes, I mean that running it both locally and once merged there shouldn't be a problem, but viewing it at https://tjscalzo.github.io/inspiring-online/ doesn't work

**Comment by TJScalzo** [comment]
Oh my... I didn't even realize it was being hosted properly. I assumed it didn't do that because of the CNAME file. Hm... I'll look into that.

Anyway, thank you for the compliment at the beginning, @Haroenv. Can you offer any revision ideas for the post that I wrote?

**Comment by Haroenv** [comment]
I've found an error. If you click on `categories` twice, you get into `/categories/categories`, which doesn't work. I'll add a line note to what should be changed.

**Comment by TJScalzo** [comment]
@Haroenv Do you mean when clicking `categories` in the menu?

**Comment by Haroenv** [comment]
Nope, I mean in `menu.html`. I've just found how to fix it.

File: _includes/menu.html [hunk 1]
```diff
@@ -1,4 +1,7 @@
 <nav>
+  {% if page.page-type %}
+  <a class="category-title" id="title-{{ page.page-type | replace: ' ', '-' | downcase }}" href="/categories/{{ page.page-type | replace: ' ', '-' | downcase }}"><h1>{{ page.page-type }}</h1></a>
```
Comment (Haroenv) [inline _includes/menu.html]: Change this line to 

```html
<a class="category-title" id="title-{{ page.page-type | replace: ' ', '-' | downcase }}" href="/categories//categories/{% if page.page-type != 'Categories'%}{{ page.page-type | replace: ' ', '-' | downcase }}{% endif %}"><h1>{{ page.page-type }}</h1></a>
```

File: _includes/menu.html [hunk 2]
```diff
@@ -278,8 +281,12 @@
...
     <li><a href="/contributors">Contributors</a></li>
+    <li><a href="/categories">Categories</a></li>
     <li><a href="https://github.com/tholman/inspiring-online#contributing" target="_blank">Join in?</a></li>
     <li><a href="/meet-inspiring-online">About</a></li>
     <li><a href="/feed.xml">Feed</a></li>
   </ul>
-</nav>
\ No newline at end of file
+  {% if page.page-type %}
+  <a class="category-mobile" id="title-{{ page.page-type | replace: ' ', '-' | downcase }}" href="/categories/{{ page.page-type | replace: ' ', '-' | downcase }}"><h1>{{ page.page-type }}</h1></a>
```
Comment (Haroenv) [inline _includes/menu.html]: Same here:

```html
<a class="category-mobile" id="title-{{ page.page-type | replace: ' ', '-' | downcase }}" href="/categories/{% if page.page-type != 'Categories'%}{{ page.page-type | replace: ' ', '-' | downcase }}{% endif %}"><h1>{{ page.page-type }}</h1></a>
```

**Comment by Haroenv** [comment]
It's also kinda weird that the index doesn't say `Inspiring Online` anymore

**Comment by TJScalzo** [comment]
Ah, the title "Categories" that shows up when on the Categories page. Thank you for pointing that out!

Also, what do you mean by the index not saying Inspiring Online?

**Comment by Haroenv** [comment]
Well, it's also in the currently online version, but the menu thing 


```
---
inspiring online 
contribute
...
```

Would look better if it said `Inspiring Online` above the line IMHO

File: _includes/menu.html (hunk 1 above)
Comment (Haroenv) [inline _includes/menu.html]: I made a typo in the previous version. It should be: 

```html
<a class="category-title" id="title-{{ page.page-type | replace: ' ', '-' | downcase }}" href="/categories/{% if page.page-type != 'Categories'%}{{ page.page-type | replace: ' ', '-' | downcase }}{% endif %}"><h1>{{ page.page-type }}</h1></a>

```

**Comment by TJScalzo** [comment]
You would just need to add `page-type: Inspiring Online`. However, that would mean we'd need to make another if statement in the snippet of code we need to change for categories. Hm... I'll see what I can do to make that work.

**Comment by Haroenv** [comment]
Yeah, it'll need another if statement then

**Comment by Haroenv** [comment]
The `href` will have to be this then

```html
{% if page.page-type == 'Inspiring Online' %}/{% else %}/categories/{% if page.page-type != 'Categories'%}{{ page.page-type | replace: ' ', '-' | downcase }}{% endif %}{% endif %}
```

**Comment by TJScalzo** [comment]
Even if I were to add the extra if statement to make the title work, there's a bit of CSS that would need to be done as well to make it look right.
screen shot 2016-05-01 at 2 28 12 pm
That's another thing to debate another day. Right now it's the time to make sure everything's working with the categories. Thank you again, @Haroenv for pointing out the issue with the categories link.

**Comment by TJScalzo** [comment]
Thank you very much for your input @Haroenv, it has been very helpful. If there is anything else, please let me know right away. Also, I believe @tholman said he might take a look at this. (?) I'll love any feedback anyone gives me. This is like my baby and I want it to be the best it can be.

**Comment by JohJakob** [comment]
@TJScalzo Your post is nice! I've made some minor fixes and improvements (see [this pull request](https://github.com/TJScalzo/inspiring-online/pull/2)), but in my opinion there's nothing that should be changed.

**Comment by TJScalzo** [comment]
As far as I can see the only thing I need to add now is the gif for the post. I will try and make it soon.

**Comment by JohJakob** [comment]
@TJScalzo I'm creating a GIF.

**Comment by JohJakob** [comment]
Categories

I hope this one is OK.

**Comment by TJScalzo** [comment]
@JohJakob Would it be possible for you to go into a post and show that you can click on the category inside the post as well?

**Comment by TJScalzo** [comment]
@JohJakob I'm perfectly fine with you making a gif, I have no experience and you seem to have a great deal. Do you mind if I give a couple more critiques?
#### Tiny little things to change:
- Make window smaller/zoom in more (it has to fit on the page while still being legible)
- Go into a post and see that you can click on the category there
- Make gif loop infinitely
- End on categories page so that it loops nicely

Other than that, it's awesome and I'm glad you've made it.

**Comment by JohJakob** [comment]
@TJScalzo You're right. I'll make a better one.

**Comment by JohJakob** [comment]
Categories

This one should work.

**Comment by vaidaslamanauskas** [comment]
Great job folks! Should tag's have # symbol in front?

**Comment by JohJakob** [comment]
@vaidaslamanauskas Do you mean the selectors in CSS?

**Comment by vaidaslamanauskas** [comment]
No no, I mean visually, like #WeirdWideWeb

**Comment by JohJakob** [comment]
@vaidaslamanauskas Hmm, I don't see any octothorps.

**Comment by vaidaslamanauskas** [comment]
Yes. My suggestion is to add them :) I think visually it is good indicator

**Comment by JohJakob** [comment]
@vaidaslamanauskas Oh, I misunderstood your sentence. I wouldn't add hashtags, but we can try it.

**Comment by TJScalzo** [comment]
@vaidaslamanauskas That seems like a cool idea. I don't have access to a computer right now so I'll test what it looks like later.

**Comment by TJScalzo** [comment]
@vaidaslamanauskas and @JohJakob If it's set up correctly the tweetbot could also use that hashtag when it tweets.  (#137)

**Comment by TJScalzo** [comment]
@vaidaslamanauskas When I added the hashtags it didn't look very nice. 
However, depending on how the tweetbot works, it should still be able to use the categories as hashtags, it would just have to add the hashtag itself. Either way, it's something to look into some other time. (#137)
#### Things to do:
- @tholman said that he might take a look. I would love his feedback if he can offer it.
- @JohJakob, would it be possible for you to crop out extra space on the edges of that gif? The text appears very small when displayed in the post and so it's hard to understand what's happening. Maybe use a smaller window size when recording/creating the gif would help.
- @JohJakob, can you also go **into** a post? Like click on the title of one of them and show the category link on the right.

I'm sorry @JohJakob that I'm being so picky about the gif. I just have this vision of what I want to happen and I want to try and stick with it. Thank you for being so flexible.

**Comment by tholman** [comment]
This all looks really great! Super into it 💃 

Happy to have it without the "#'s" before each category, it feels fairly obvious that they stand apart.

Love the colors.

Your call with the gif!

**Comment by TJScalzo** [comment]
Thank you for chiming in @tholman. Everything is now basically done, I just hope @JohJakob has the time to edit the gif so that this can be published!

**Comment by JohJakob** [comment]
@TJScalzo No problem! I'll do it soon.

**Comment by tholman** [comment]
💃 This has been a great effort!

**Comment by TJScalzo** [comment]
## I'm just posting here to remind myself to change the date of the post once @JohJakob creates and adds the gif.

**Comment by Haroenv** [comment]
I'm uploading a gif currently

**Comment by Haroenv** [comment]
https://github.com/TJScalzo/inspiring-online/pull/4

**Comment by TJScalzo** [comment]
### Okay!
This is finished! This is ready to be merged once someone with write access sees this!

**Comment by Haroenv** [comment]
Maybe wait for the best time of the day? 12 ET or something was ideal I thought. Anyway; :shipit:

**Comment by TJScalzo** [comment]
### Okay!
This is finished! I'd love it if someone with write access could merge this at a reasonable time of day (maybe 12 ET as suggested by @Haroenv).

**Comment by JohJakob** [comment]
Shall I merge this pull request?

**Comment by TJScalzo** [comment]
@JohJakob Please merge this the next chance you get!

**Comment by JohJakob** [comment]
🎉

**Pull request merged**